tests/phpunit/tests/rest-api/wpRestTemplatesController.php

```diff
@@ -845,4 +845,46 @@ public function test_get_template_fallback() {
 		$response = rest_get_server()->dispatch( $request );
 		$this->assertSame( 'index', $response->get_data()['slug'], 'Should fallback to `index.html` when  ignore_empty is `true`.' );
 	}
+
+	/**
+	 * @ticket 57851
+	 *
+	 * @covers WP_REST_Templates_Controller::prepare_item_for_database
+	 */
+	public function test_prepare_item_for_database() {
+		$endpoint = new WP_REST_Templates_Controller( 'wp_template_part' );
+
+		$prepare_item_for_database = new ReflectionMethod( $endpoint, 'prepare_item_for_database' );
+		$prepare_item_for_database->setAccessible( true );
+
+		$body_params = array(
+			'title'   => 'Untitled Template Part',
+			'slug'    => 'untitled-template-part',
+			'content' => '',
+		);
+
+		$request = new WP_REST_Request( 'POST', '/wp/v2/template-parts' );
+		$request->set_body_params( $body_params );
+
+		$prepared = $prepare_item_for_database->invoke( $endpoint, $request );
+
+		$this->assertInstanceOf( 'stdClass', $prepared );
+
+		$prepared_array = (array) $prepared;
```

Comment:
I think we should be able to just use `$prepared` here and assert using this:
```php
$this->assertObjectHasAttribute(
	'post_type',
	$prepared,
	'The "post_type" was not included in the prepared template part.'
);
```

Reply:
@costdev 

I didn't use `assertObjectHasAttribute` since this assertion was removed in PHPUnit 10.

Reference:
- https://github.com/sebastianbergmann/phpunit/issues/4601

Reply:
I don't believe the core test suite supports PHPUnit 10+ at this time. When that happens, it will most likely make use of [PHPUnit Polyfills 2.0.0](https://github.com/Yoast/PHPUnit-Polyfills/releases/tag/2.0.0) which includes [polyfills for the new assertions](https://github.com/Yoast/PHPUnit-Polyfills/pull/116) and perform a replacement on `assertObjectHasAttribute(` / `assertObjectNotHasAttribute(` across all tests. Using these now makes it easier to adjust the tests for PHPUnit 10+ later.

Reply:
Got it. I updated the tests. Thanks.